### PR TITLE
nixos/cadvisor: expose interval options

### PR DESCRIPTION
**Pull request by tilpner**
###### Motivation for this change

Options are described here: https://github.com/google/cadvisor/blob/master/docs/runtime_options.md

If Prometheus only queries every 10s, there's no point for cadvisor to query the system every second. These options allow to disable that unnecessary work.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by tilpner** [comment]
cc @basvandijk @offlinehacker

**Comment by tilpner** [comment]
These are `nullOr str` so that we don't hardcode their defaults in nixpkgs. I would usually choose a nested structure for shorter names, but that didn't fit with the rest of the options.

File: nixos/modules/services/monitoring/cadvisor.nix [hunk 1]
```diff
@@ -84,6 +84,27 @@ in {
         type = types.bool;
         description = "Cadvisor storage driver, enable secure communication.";
       };
+
+      globalHousekeepingInterval = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "Interval between global housekeepings";
```
Comment (Infinisil) [inline nixos/modules/services/monitoring/cadvisor.nix]: Nitpick: Descriptions should have a `.` at the end

File: nixos/modules/services/monitoring/cadvisor.nix [hunk 2]
```diff
@@ -84,6 +84,27 @@ in {
...
+        example = "1m0s";
```
Comment (Infinisil) [inline nixos/modules/services/monitoring/cadvisor.nix]: Can you put the format for this in the description?

**Review by infinisil — CHANGES_REQUESTED** [review]

**Review by infinisil — APPROVED** [review]

**Comment by Infinisil** [comment]
Hmm although, if there are even more options in the CLI, it might make more sense to just have an `extraOptions` NixOS option to pass additional CLI options.

**Comment by tilpner** [comment]
@Infinisil That's a style question, for which there doesn't seem to be a consensus. I'd be in favor of adding less structured catch-all options (`attrs` usually), but that's out of scope for this PR.

**Comment by Infinisil** [comment]
Well I'm very much in favor of that, see NixOS/rfcs#42, advantages are listed there

**Comment by tilpner** [comment]
@Infinisil You don't have to convince me, I already agree. But that's not for this PR, and I probably won't send another one for a few days. Close this PR if you feel an unstructured option will be accepted and is reasonably easy to implement.

**Comment by Infinisil** [comment]
I think it would be rather easy to implement, but I don't mind this PR either

**Comment by tilpner** [comment]
How do we proceed with this?

**Pull request closed without merging**